File: web-forms/src/store/evaluation/condicoesPele.js
import { defineStore } from 'pinia';
import { ref } from 'vue';

const curativoBase = {
    local_curativo: '',
    tipo_cobertura: '',
    tamanho_curativo: '',
    qtd_curativo: '',
    frequencia_troca: '',
    materiais_curativo: ''
};

export const useCondicoesPeleStore = defineStore('condicoesPele', () => {

    const condicoesPele = ref({
        condicao_cutanea_mucosa: null,
        realiza_curativo: false,
        curativos: []
    });

    function adicionarCurativo() {

        condicoesPele.value.curativos.push({ ...curativoBase });
    }

    function removerCurativo(index) {
        if (condicoesPele.value.curativos.length > 0) {
            condicoesPele.value.curativos.splice(index, 1);
        }
    }

    function reset() {
        condicoesPele.value = JSON.parse(JSON.stringify({
            condicao_cutanea_mucosa: null,
            realiza_curativo: false,
            curativos: []
        }))
    }

    return {
        condicoesPele,
        adicionarCurativo,
        removerCurativo,
        reset
    };
}, {
    persist: true
});

File: backend/internal/avaliation/domain/condicoes_pele.go
package domain

type CondicoesPele struct {
	CondicaoCutaneaMucosa string     `json:"condicao_cutanea_mucosa"`
	RealizaCurativo       bool       `json:"realiza_curativo"`
	Curativos             []Curativo `json:"curativos"`
}

type Curativo struct {
	LocalCurativo     string `json:"local_curativo"`
	TipoCobertura     string `json:"tipo_cobertura"`
	TamanhoCurativo   string `json:"tamanho_curativo"`
	QtdCurativo       string `json:"qtd_curativo"`
	FrequenciaTroca   string `json:"frequencia_troca"`
	MateriaisCurativo string `json:"materiais_curativo"`
}
